packages/editor/src/components/post-actions/set-as-homepage.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useEffect, useMemo, useState } from '@wordpress/element';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useDispatch, useSelect, resolveSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from '../../utils/get-item-title';
+import { unlock } from '../../lock-unlock';
+
+const SetAsHomepageModal = ( { items, closeModal, onActionPerformed } ) => {
+	const [ item ] = items;
+	const pageTitle = getItemTitle( item );
+	const { showOnFront, currentHomePage } = useSelect( ( select ) => {
+		const { getEntityRecord } = select( coreStore );
+		const { getHomePage } = unlock( select( coreStore ) );
+		const siteSettings = getEntityRecord( 'root', 'site' );
+		const homePage = getHomePage();
+		return {
+			showOnFront: siteSettings?.show_on_front,
+			currentHomePage: homePage.postId,
+		};
+	} );
+	const [ currentHomePageTitle, setCurrentHomePageTitle ] = useState( '' );
+	const isPageDraft = item.status === 'draft';
+
+	const { getEntityRecord } = resolveSelect( coreStore );
+	const { saveEditedEntityRecord, saveEntityRecord } =
+		useDispatch( coreStore );
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+
+	useEffect( () => {
+		const handleGetCurrentHomepage = async () => {
+			const currentHomePageItem = await getEntityRecord(
+				'postType',
+				'page',
+				currentHomePage
+			);
+			setCurrentHomePageTitle( getItemTitle( currentHomePageItem ) );
+		};
+		handleGetCurrentHomepage();
+	}, [ currentHomePage, getEntityRecord ] );
```

Comment:
I think we should use the standard data layer selectors and avoid localy state.

We can default the UI to show an empty string for the title. Let's make sure we account for the possibility of the selector returning `undefined`.

Reply:
I've refactored how the current homepage record is fetched in https://github.com/WordPress/gutenberg/pull/65426/commits/9685c3a7e877a5d3f408d4ec03004376154b192f and I think it addresses this comment, as it no longer requires a `useEffect`.